### PR TITLE
add spell check for news and readme

### PR DESCRIPTION
**Pull request by vsoch**
Problem: the NEWS.md is not spell checked, and goes out to releases.
Solution: add a strong spell checker to be run on CI so we catch issues.

### CODE REVIEW

**Comment by chu11** [comment]
similar to https://github.com/flux-framework/flux-core/pull/5074 I'll try to fix up a ton of the typos in here too.  There's a lot more false positives than  in flux-core, so will have to go through it a little bit one by one :-(  Don't have to necessarily fix all of them, just a good chunk.

**Comment by vsoch** [comment]
Sounds good - thanks @chu11 ! Please push / edit this branch as you see fit.

**Comment by chu11** [comment]
@vsoch similar the fixup in flux-core, could the commit be split into a typo fix commit and a CI fix.  Thanks.

**Comment by vsoch** [comment]
All set.

**Comment by chu11** [comment]
i think you forgot to tweak title of the `add spell check for news and readme` commit.  Perhaps just "NEWS.md: fix some typos" :-)

**Comment by vsoch** [comment]
I didn’t forget, I thought it was OK. I think nit picky stuff like this is a waste of time, but since you asked I guess I have to change it!

**Comment by grondo** [comment]
I think @vsoch has to add you as collaborator to researchapps...

**Comment by vsoch** [comment]
Done! I invited all of you.

File: resource/traversers/dfu_impl.cpp [hunk 1]
```diff
@@ -293,7 +293,7 @@ const std::vector<Resource> &dfu_impl_t::test (vtx_t u,
      * For this case, when you visit the "node" resource vertex, the next
      * Jobspec resource that should be used at the next recursion should
      * be socket[1]. And we enable this if pristine is true.
-     * But then once the first match is made, any mis-mismatch afterwards
+     * But then once the first match is made, any miss-mismatch afterwards
```
Comment (jameshcorbett) [inline resource/traversers/dfu_impl.cpp:296]: I don't think I understand this either way...

**Review by jameshcorbett — APPROVED** [review]
LGTM!

File: resource/traversers/dfu_impl.cpp (hunk 1 above)
Comment (grondo) [reply resource/traversers/dfu_impl.cpp:296]: The typo software stumbled here, seems like it should be "mismatch"

Comment (chu11) [reply resource/traversers/dfu_impl.cpp:296]: doh, i'll fix this up

**Comment by chu11** [comment]
fixed up that one remaining typo, force pushed it, hope that's ok!

setting MWP

**Pull request merged**